compiler/src/iree/compiler/Codegen/Common/TransformExtensions/CommonExtensions.cpp

```diff
@@ -1083,5 +1083,45 @@ void transform_dialect::GpuDistributeSharedMemoryCopyOp::getEffects(
   transform::modifiesPayload(effects);
 }
 
+//===----------------------------------------------------------------------===//
+// CreateTileSizesOp
+//===----------------------------------------------------------------------===//
+
+DiagnosedSilenceableFailure transform_dialect::CreateTileSizesOp::applyToOne(
+    transform::TransformRewriter &rewriter, Operation *target,
+    transform::ApplyToEachResultList &results,
+    transform::TransformState &state) {
+  Builder builder(getTarget().getContext());
+  ArrayRef<int64_t> shape =
+      llvm::cast<ShapedType>(target->getResult(0).getType()).getShape();
```

Comment:
`emitDefiniteFailure()` if the target does not have exactly one result and the return type isn't `ShapedType`.